File: apps/core/eave/core/graphql/types/address.py
import strawberry

from eave.core.lib.address import Address, BaseAddress, format_address


@strawberry.type(name="Address")
class GraphQLAddress(BaseAddress):
    @strawberry.field
    def formatted_multiline(self) -> str:
        return format_address(self, singleline=False)

    @strawberry.field
    def formatted_singleline(self) -> str:
        return format_address(self, singleline=True)

    @classmethod
    def from_address(cls, address: Address) -> "GraphQLAddress":
        """
        Convert a plain Address object to a GraphQLAddress object
        """
        return GraphQLAddress(
            address1=address.address1,
            address2=address.address2,
            city=address.city,
            state=address.state,
            zip_code=address.zip_code,
            country=address.country,
        )

    def to_address(self) -> Address:
        """
        Convert a GraphQLAddress object to a plain Address object.
        """
        return Address(
            address1=self.address1,
            address2=self.address2,
            city=self.city,
            state=self.state,
            zip_code=self.zip_code,
            country=self.country,
        )

File: apps/dashboard/eave/dashboard/js/util/address.ts
interface Address {
  street?: string;
  city?: string;
  state?: string;
  zipCode?: string;
}

export function parseAddress(formattedAddress?: string | null): Address {
  const address: Address = {};
  if (formattedAddress) {
    const addressParts = formattedAddress.split(", ");
    address.street = addressParts[0];
    address.city = addressParts[1];
    address.state = addressParts[2];
    address.zipCode = addressParts[3];
  }
  return address;
}
